web/src/connection/api/index.tsx

```diff
@@ -15,3 +17,117 @@ export const cadastro = async (nome: string, email: string, senha: string) => {
 
   return response.ok;
 };
+
+export const login = async (email: string, senha: string) => {
+  const response = await fetch('http://localhost:8090/users/auth', {
+    method: 'POST',
+    body: JSON.stringify({
+      email: email,
+    }),
+    headers: {
+      'Content-type': 'application/json',
+    },
+  });
+
+  if (!response.ok) {
+    return;
+  }
+  //guardar o salt em uma variável
+  const salt = await response.json();
+
+  //Calcular o primeiro hash com a string fixa do cadastro
+  const primeiroHash = sha512(senha + 'justaTI');
+
+  //Calcular a segunda hash concatenando a primeira hash com o salt
+  const segundoHash = sha512(primeiroHash + salt);
+
+  //Fazer requisição ao segundo endpoint de autenticação
+
+  const secondResponse = await fetch('http://localhost:8090/users/auth', {
+    method: 'PUT',
+    body: JSON.stringify({
+      email: email,
+      secondHash: segundoHash,
+    }),
+    headers: {
+      'Content-type': 'application/json',
+    },
+  });
+
+  if (!secondResponse.ok) {
+    return;
+  }
+
+  //Retornar a resposta da requisição que é o JWT
+  return await secondResponse.json();
+};
+
+export const getUser = async () => {
+  const {jwt, setJwt} = useContext(Context);
+  const response = await fetch('http://localhost:8090/users/me', {
+    method: 'GET',
+    headers: {
+      'Content-type': 'application/json',
+      Authorization: jwt,
+    },
+  });
+
+  return await response.json();
+};
+
+export type Note = {title: string; body: string; id: string};
+
+export const useApi = () => {
+  const {jwt, setJwt} = useContext(Context);
+  return {
+    getNotes: async (): Promise<Note[]> => {
+      const response = await fetch('http://localhost:8090/notes', {
+        method: 'GET',
+        headers: {
+          Authorization: jwt,
+        },
+      });
+      return await response.json();
+    },
+    patchNotes: async (note: Note): Promise<Note> => {
```

Comment:
This method is for editing a single note, right? Shouldn't it be `pathNote`, singular?